edgedb-client/src/builder.rs

```diff
@@ -965,7 +970,9 @@ impl Builder {
             self._connect_with_cert_verifier(Arc::new(tls::NullVerifier)).await
         } else {
             let verify_host = self.do_verify_hostname();
-            if verify_host && IpAddr::from_str(&self.host).is_ok() {
+            if verify_host && DNSNameRef::try_from_ascii_str(
```

Comment:
Can you briefly describe what's going on here? IpAddr should parse IPv6. And if we can't parse `DNSNameRef` in may mean anything like unicode things in hostname, not just IP address, so we should make another error message.

Reply:
The reason to make this change was to be consistent with the other `IpAddr::from_str` change below, which is because `std::net::IpAddr` won't parse scoped IPv6 address (`std::net::ToSocketAddrs` does, but doesn't distinguish between IP address and DNS names), so the CLI thinks scoped addresses are safe DNS names, but `DNSNameRef` doesn't think so.

As for this change, I think it makes sense to revert it so that it reports the right error in all cases but one - specifying a scoped address but not providing a root cert, when the CLI reports `InvalidDNSNameError` it actually should report "cannot use system root certificates with an IP address", but it's probably okay because it's such a rare case.